internal/report/report.go

```diff
@@ -433,7 +433,19 @@ func PrintAssembly(w io.Writer, rpt *Report, obj plugin.ObjTool, maxFuncs int) e
 	}
 
 	if len(syms) == 0 {
-		return fmt.Errorf("no matches found for regexp: %s", o.Symbol)
+		if address == nil { // User provides a symbol instead of an address
+			if len(symbols) == 0 {
+				return fmt.Errorf("no matches found for regexp %s in binaries", o.Symbol)
+			} else {
+				return fmt.Errorf("regex %s found in binaries, but not in the profile", o.Symbol)
+			}
+		} else { // User provides an address
+			if len(symbols) == 0 {
+				return fmt.Errorf("the address 0x%x is not found in binaries", *address)
```

Comment:
nit: "the" is used here, but not in the message below, worth being consistent. I would remove it for brevity.

Reply:
Done.